.github/workflows/static-analysis.yaml

```diff
@@ -30,12 +30,33 @@ jobs:
 
       - name: Set up Python
         uses: actions/setup-python@v5
+        id: setup_python
         with:
           python-version: 3.9
 
-      - name: Install pre-commit
+      - name: UV Cache
+        # Manually cache the uv cache directory
+        # until setup-python supports it:
+        # https://github.com/actions/setup-python/issues/822
+        uses: actions/cache@v4
+        id: cache-uv
+        with:
+          path: ~/.cache/uv
+          key: uvcache-${{ runner.os }}-${{ steps.setup_python.outputs.python-version }}-${{ hashFiles('requirements-client.txt', 'requirements.txt', 'requirements-dev.txt') }}
+
+      - name: Install packages
         run: |
-          pip install pre-commit
+          python -m pip install -U uv pre-commit
+          uv pip install --upgrade --system -e .[dev]
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache-dependency-path: "**/package-lock.json"
+
+      - name: Install mintlify-scrape
+        run: npm install -g @mintlify/scraping
```

Comment:
Do you need to install mintlify too? Not sure if it's now in our required development packages.

Reply:
For this particular script, we only need `mintlify-scrape`